source/DSMainWindow.cpp

```diff
@@ -98,6 +102,36 @@ DSMainWindow::DSMainWindow(QWidget* parent):
     setWindowTitle("Despeect");
 }
 
+void DSMainWindow::createActionLoadVoice() {
+    const QIcon openIcon = QIcon::fromTheme("document-open", QIcon(":/images/open.png"));
+
+    QAction *openAct = new QAction(openIcon, QObject::tr("Load Configuration File"), this);
+    openAct->setShortcuts(QKeySequence::Open);
+    openAct->setStatusTip(tr("Load Configuration File"));
+    QObject::connect(openAct,
+                     &QAction::triggered,
+                     [=] () {
+                             QString path = QFileDialog::getOpenFileName(this,QObject::tr("Oper Configuration File"), "../../", QObject::tr("Voice Files (*.json)"));
+                             voice_path = path.toStdString();
+                             adapter->loadVoice(voice_path);
+                        });
+    actions.push_back(openAct);
+}
+
+void DSMainWindow::createActionShowVoicePath() {
+    const QIcon openIcon = QIcon::fromTheme("dialog-information");
+    QAction *openAct = new QAction(openIcon, QObject::tr("Show Voice Path"), this);
+    openAct->setStatusTip(tr("Show Voice Path"));
+    QObject::connect(openAct,
```

Comment:
Anche qui al posto che nella lambda metterei il codice in uno slot dentro DSMainWindow. Poi non serve qualificare la funzione connect, non viene mai ridefinita, e tutte le classi derivano da QObject